### PR TITLE
Add test for `createBufferedTransformStream`

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
Adds a simple test for `createBufferedTransformStream`.

Wasn't sure what other scenarios should be tested. Open to ideas @gnoff & @wyattjoh 

Closes NEXT-2749

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: efd71598cb02bae2edf9f360691a5f9dee93a3d1

`TURBOPACK=1 pnpm test test/integration/create-next-app/package-manager.test.ts` (turbopack)
-  > should use npm as the package manager on supplying --use-npm with example
-  > should use Yarn as the package manager on supplying --use-yarn with example
-  > should use pnpm as the package manager on supplying --use-pnpm with example
-  > should use Bun as the package manager on supplying --use-bun with example
-  > should infer npm as the package manager with example
-  > should infer yarn as the package manager with example
-  > should infer pnpm as the package manager with example
-  > should infer Bun as the package manager with example

<details>
<summary>Expand output</summary>

  ● should use npm as the package manager on supplying --use-npm with example

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      80 |   for (const file of files) {
      81 |     try {
    > 82 |       expect(existsSync(resolve(projectRoot, file))).toBe(true)
         |                                                      ^
      83 |     } catch (err) {
      84 |       require('console').error(
      85 |         `missing expected file ${file}`,

      at toBe (integration/create-next-app/lib/utils.ts:82:54)
      at integration/create-next-app/package-manager.test.ts:88:28
      at useTempDir (lib/use-temp-dir.ts:24:5)
      at Object.<anonymous> (integration/create-next-app/package-manager.test.ts:72:3)

  ● should use Yarn as the package manager on supplying --use-yarn with example

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      80 |   for (const file of files) {
      81 |     try {
    > 82 |       expect(existsSync(resolve(projectRoot, file))).toBe(true)
         |                                                      ^
      83 |     } catch (err) {
      84 |       require('console').error(
      85 |         `missing expected file ${file}`,

      at toBe (integration/create-next-app/lib/utils.ts:82:54)
      at integration/create-next-app/package-manager.test.ts:166:28
      at useTempDir (lib/use-temp-dir.ts:24:5)
      at Object.<anonymous> (integration/create-next-app/package-manager.test.ts:150:3)

  ● should use pnpm as the package manager on supplying --use-pnpm with example

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      80 |   for (const file of files) {
      81 |     try {
    > 82 |       expect(existsSync(resolve(projectRoot, file))).toBe(true)
         |                                                      ^
      83 |     } catch (err) {
      84 |       require('console').error(
      85 |         `missing expected file ${file}`,

      at toBe (integration/create-next-app/lib/utils.ts:82:54)
      at integration/create-next-app/package-manager.test.ts:244:28
      at useTempDir (lib/use-temp-dir.ts:24:5)
      at Object.<anonymous> (integration/create-next-app/package-manager.test.ts:228:3)

  ● should use Bun as the package manager on supplying --use-bun with example

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      80 |   for (const file of files) {
      81 |     try {
    > 82 |       expect(existsSync(resolve(projectRoot, file))).toBe(true)
         |                                                      ^
      83 |     } catch (err) {
      84 |       require('console').error(
      85 |         `missing expected file ${file}`,

      at toBe (integration/create-next-app/lib/utils.ts:82:54)
      at integration/create-next-app/package-manager.test.ts:318:28
      at useTempDir (lib/use-temp-dir.ts:24:5)
      at Object.<anonymous> (integration/create-next-app/package-manager.test.ts:302:3)

  ● should infer npm as the package manager with example

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      80 |   for (const file of files) {
      81 |     try {
    > 82 |       expect(existsSync(resolve(projectRoot, file))).toBe(true)
         |                                                      ^
      83 |     } catch (err) {
      84 |       require('console').error(
      85 |         `missing expected file ${file}`,

      at toBe (integration/create-next-app/lib/utils.ts:82:54)
      at integration/create-next-app/package-manager.test.ts:389:28
      at useTempDir (lib/use-temp-dir.ts:24:5)
      at Object.<anonymous> (integration/create-next-app/package-manager.test.ts:366:3)

  ● should infer yarn as the package manager with example

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      80 |   for (const file of files) {
      81 |     try {
    > 82 |       expect(existsSync(resolve(projectRoot, file))).toBe(true)
         |                                                      ^
      83 |     } catch (err) {
      84 |       require('console').error(
      85 |         `missing expected file ${file}`,

      at toBe (integration/create-next-app/lib/utils.ts:82:54)
      at integration/create-next-app/package-manager.test.ts:472:28
      at useTempDir (lib/use-temp-dir.ts:24:5)
      at Object.<anonymous> (integration/create-next-app/package-manager.test.ts:449:3)

  ● should infer pnpm as the package manager with example

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      80 |   for (const file of files) {
      81 |     try {
    > 82 |       expect(existsSync(resolve(projectRoot, file))).toBe(true)
         |                                                      ^
      83 |     } catch (err) {
      84 |       require('console').error(
      85 |         `missing expected file ${file}`,

      at toBe (integration/create-next-app/lib/utils.ts:82:54)
      at integration/create-next-app/package-manager.test.ts:555:28
      at useTempDir (lib/use-temp-dir.ts:24:5)
      at Object.<anonymous> (integration/create-next-app/package-manager.test.ts:532:3)

  ● should infer Bun as the package manager with example

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      80 |   for (const file of files) {
      81 |     try {
    > 82 |       expect(existsSync(resolve(projectRoot, file))).toBe(true)
         |                                                      ^
      83 |     } catch (err) {
      84 |       require('console').error(
      85 |         `missing expected file ${file}`,

      at toBe (integration/create-next-app/lib/utils.ts:82:54)
      at integration/create-next-app/package-manager.test.ts:630:28
      at useTempDir (lib/use-temp-dir.ts:24:5)
      at Object.<anonymous> (integration/create-next-app/package-manager.test.ts:607:3)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js test-createBufferedTransformStream | Change |
| - | - | - | - |
| buildDuration | 13.6s | 13.8s | ⚠️ +186ms |
| buildDurationCached | 7.3s | 6.2s | N/A |
| nodeModulesSize | 198 MB | 198 MB | ⚠️ +20.5 kB |
| nextStartRea..uration (ms) | 430ms | 431ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js test-createBufferedTransformStream | Change |
| - | - | - | - |
| 2453-HASH.js gzip | 30.5 kB | 30.5 kB | N/A |
| 3304.HASH.js gzip | 181 B | 181 B | ✓ |
| 3f784ff6-HASH.js gzip | 53.7 kB | 53.7 kB | N/A |
| 8299-HASH.js gzip | 5.04 kB | 5.04 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 242 B | 241 B | N/A |
| main-HASH.js gzip | 32.2 kB | 32.2 kB | N/A |
| webpack-HASH.js gzip | 1.68 kB | 1.68 kB | N/A |
| Overall change | 45.4 kB | 45.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js test-createBufferedTransformStream | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js test-createBufferedTransformStream | Change |
| - | - | - | - |
| _app-HASH.js gzip | 196 B | 197 B | N/A |
| _error-HASH.js gzip | 184 B | 184 B | ✓ |
| amp-HASH.js gzip | 505 B | 505 B | ✓ |
| css-HASH.js gzip | 324 B | 325 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 258 B | 258 B | ✓ |
| head-HASH.js gzip | 352 B | 352 B | ✓ |
| hooks-HASH.js gzip | 370 B | 371 B | N/A |
| image-HASH.js gzip | 4.21 kB | 4.21 kB | ✓ |
| index-HASH.js gzip | 259 B | 259 B | ✓ |
| link-HASH.js gzip | 2.67 kB | 2.67 kB | N/A |
| routerDirect..HASH.js gzip | 314 B | 312 B | N/A |
| script-HASH.js gzip | 386 B | 386 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 6.57 kB | 6.57 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js test-createBufferedTransformStream | Change |
| - | - | - | - |
| _buildManifest.js gzip | 481 B | 484 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js test-createBufferedTransformStream | Change |
| - | - | - | - |
| index.html gzip | 529 B | 528 B | N/A |
| link.html gzip | 541 B | 540 B | N/A |
| withRouter.html gzip | 524 B | 523 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js test-createBufferedTransformStream | Change |
| - | - | - | - |
| edge-ssr.js gzip | 95.1 kB | 95.1 kB | N/A |
| page.js gzip | 3.06 kB | 3.07 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js test-createBufferedTransformStream | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 628 B | 623 B | N/A |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 25.5 kB | 25.5 kB | N/A |
| edge-runtime..pack.js gzip | 839 B | 839 B | ✓ |
| Overall change | 990 B | 990 B | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js test-createBufferedTransformStream | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 171 kB | 171 kB | ✓ |
| app-page-exp..prod.js gzip | 96.5 kB | 96.5 kB | ✓ |
| app-page-tur..prod.js gzip | 98.2 kB | 98.2 kB | ✓ |
| app-page-tur..prod.js gzip | 92.7 kB | 92.7 kB | ✓ |
| app-page.run...dev.js gzip | 149 kB | 149 kB | ✓ |
| app-page.run..prod.js gzip | 91.2 kB | 91.2 kB | ✓ |
| app-route-ex...dev.js gzip | 21.3 kB | 21.3 kB | ✓ |
| app-route-ex..prod.js gzip | 15 kB | 15 kB | ✓ |
| app-route-tu..prod.js gzip | 15 kB | 15 kB | ✓ |
| app-route-tu..prod.js gzip | 14.8 kB | 14.8 kB | ✓ |
| app-route.ru...dev.js gzip | 20.9 kB | 20.9 kB | ✓ |
| app-route.ru..prod.js gzip | 14.8 kB | 14.8 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.52 kB | 9.52 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.8 kB | 9.8 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.52 kB | 9.52 kB | ✓ |
| pages-turbo...prod.js gzip | 22.3 kB | 22.3 kB | ✓ |
| pages.runtim...dev.js gzip | 22.9 kB | 22.9 kB | ✓ |
| pages.runtim..prod.js gzip | 22.3 kB | 22.3 kB | ✓ |
| server.runti..prod.js gzip | 50.5 kB | 50.5 kB | ✓ |
| Overall change | 947 kB | 947 kB | ✓ |

</details>

<details>
<summary><strong>build cache</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js test-createBufferedTransformStream | Change |
| - | - | - | - |
| 0.pack gzip | 1.56 MB | 1.56 MB | ⚠️ +1.29 kB |
| index.pack gzip | 105 kB | 105 kB | N/A |
| Overall change | 1.56 MB | 1.56 MB | ⚠️ +1.29 kB |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>middleware.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>edge-ssr.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: efd71598cb02bae2edf9f360691a5f9dee93a3d1

File: packages/next/src/server/stream-utils/node-web-streams-helper.test.ts [hunk 1]
```diff
@@ -178,7 +179,28 @@ describe('node-web-stream-helpers', () => {
...
+      const stream = createBufferedTransformStream()
+      const input = new ReadableStream({
+        start(controller) {
+          // enquque 3 chunks of data
+          controller.enqueue(new Uint8Array([97, 98, 99, 100]))
+          controller.enqueue(new Uint8Array([101, 102, 103, 104]))
+          controller.enqueue(new Uint8Array([105, 106, 107, 108]))
+          controller.close()
+        },
+      })
```
Comment (gnoff) [inline packages/next/src/server/stream-utils/node-web-streams-helper.test.ts:193]: I think a more complex test would be something like

```
const bigChunks = [
  new Uint8Array(... some 15 bytes or so)
  new Uint8Array(... some 15 bytes or so)
]

const input = new ReadableStream({
  async pull() {
    controller.enqueue(bigChunks[0].slice(0, 5))
    // wait a microtask
    await Promise.resolve()
    controller.enqueue(bigChunks[0].slice(5, 10))
    // wait a microtask
    controller.enqueue(bigChunks[0].slice(10))
    
    // Now wait a Task
    await new Promise(r => setTimeout(r, 0))
    
    controller.enqueue(bigChunks[1].slice(0, 5))
    // wait a microtask
    await Promise.resolve()
    controller.enqueue(bigChunks[1].slice(5, 10))
    // wait a microtask
    controller.enqueue(bigChunks[1].slice(10))
    
    controller.close()
  }
})
```

Then you assert the output chunks equal `bigChunks`

**Review by gnoff — APPROVED** [review]
Its a good test but consider exercising it a bit more

**Review by gnoff — APPROVED** [review]

**Pull request merged**